app/src/userpages/components/StreamPage/Show/index.jsx

```diff
@@ -264,11 +241,67 @@ export class StreamShowView extends Component<Props, State> {
     }
 }
 
+function StreamLoader(props: Props) {
+    const isMounted = useIsMounted()
+    const propsRef = useRef(props)
+    propsRef.current = props
+    const { id: streamId } = props.match.params || {}
+
+    const initStream = useCallback(async (id: StreamId) => {
+        let { current: currentProps } = propsRef
+        await propsRef.current.getKeys()
+        if (!isMounted()) { return }
+        currentProps = propsRef.current
+        return Promise.all([
+            currentProps.getStream(id).then(() => {
+                if (!isMounted()) { return }
+                currentProps.openStream(id)
+                currentProps.initEditStream()
+            }),
+            currentProps.getMyStreamPermissions(id),
+        ])
+    }, [isMounted, propsRef])
```

Comment:
Isn't this the same as using `props` instead of `propsRef `? Or it's like below to store refs to actions?

Reply:
It's actually that we don't care about the props changing… always just use whatever the latest value is. Basically the intention here is to as closely as possible match how it was working in the class when we were accessing callbacks via `this.props` after an async action, rather than caching the value of the callback(s) from `this.props` at invocation time.

I think in this case it probably doesn't matter really because I don't think redux will change the identities of anything in the current `mapDispatchToProps` config, but if we later adjust how mapDispatchToProps is initialised and the function identities do change in a meaningful way, then it could lead to a subtly broken sequence, where it's using the old version of a prop long after it was updated. 

It is also questionable whether it makes more sense to continue a sequence using the props as they were defined in the initial invocation, or allow a mix old and new props like this, but I intended to port the class behaviour as closely as possible so that's why it ended up looking like so.